test/jdk/javax/swing/border/Test4129681.java

```diff
@@ -21,39 +21,65 @@
  * questions.
  */
 
-/*
- * @test
- * @bug 4129681
- * @summary Tests enabling/disabling of titled border's caption
- * @author Sergey Malenkov
- * @run applet/manual=yesno Test4129681.html
- */
-
 import java.awt.BorderLayout;
 import java.awt.event.ItemEvent;
-import java.awt.event.ItemListener;
 import javax.swing.BorderFactory;
-import javax.swing.JApplet;
 import javax.swing.JCheckBox;
+import javax.swing.JFrame;
 import javax.swing.JLabel;
+import javax.swing.SwingUtilities;
 
-public class Test4129681 extends JApplet implements ItemListener {
-    private JLabel label;
+/*
+ * @test
+ * @bug 4129681
+ * @summary Tests enabling/disabling of titled border's caption
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual Test4129681
+ */
 
-    @Override
-    public void init() {
-        JCheckBox check = new JCheckBox("disable");
-        check.addItemListener(this);
+public class Test4129681 {
+    private static JLabel label;
+    private static JFrame frame;
 
-        this.label = new JLabel("message");
-        this.label.setBorder(BorderFactory.createTitledBorder("label"));
-        this.label.setEnabled(!check.isSelected());
+    public static void main(String[] args) throws Exception {
+        String testInstructions = """
+                When applet starts, you'll see a checkbox
+                and a label with a titled border.
+                Turn on the checkbox to disable the label.
+                The test passes if the title of the border
+                is disabled as well as the label.
+                """;
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .title("JInternalFrame Instructions")
+                .instructions(testInstructions)
+                .testTimeOut(5)
+                .rows(6)
+                .columns(35)
+                .build();
 
-        add(BorderLayout.NORTH, check);
-        add(BorderLayout.CENTER, this.label);
+        SwingUtilities.invokeAndWait(() -> {
+            init();
+        });
+        passFailJFrame.awaitAndCheck();
     }
 
-    public void itemStateChanged(ItemEvent event) {
-        this.label.setEnabled(ItemEvent.DESELECTED == event.getStateChange());
+    public static void init() {
```

Comment:
I propose using `testUI` method of the builder, it hides the complexity of registering and positioning the frame. Modify this method to return the created `frame`.

See PR #17838 as an example or other PRs. The javadoc for `PassFailJFrame` also contains the recommended usage.